modules/table/src/main/java/org/apache/ignite/internal/table/TableImpl.java

```diff
@@ -361,33 +348,21 @@ public void registerSortedIndex(
     public void unregisterIndex(UUID indexId) {
         indexLockerFactories.remove(indexId);
         indexStorageAdapterFactories.remove(indexId);
+
+        CompletableFuture<?> indexToWaitFuture = indexesToWait.remove(indexId);
+
+        if (indexToWaitFuture != null) {
+            indexToWaitFuture.complete(null);
```

Comment:
Do we test this behavior anywhere? Shouldn't this future be completed with an exception? How does this scenario work?

Reply:
No, right now the deletion of the index is not implemented correctly in general and not tested as it should, I will create a ticket for this.

Reply:
Please add a TODO here